test_scripts/Capabilities/PersistingHMICapabilities/032_SDL_request_capability_in_case_is_not_present_in_cache_and_default_files.lua

```diff
@@ -0,0 +1,97 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal:https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0249-Persisting-HMI-Capabilities-specific-to-headunit.md
+--
+-- Description: Check that the SDL sends appropriate HMI capabilities (VR/TTS/RC/UI/Buttons/VehicleInfo etc) request
+--  to HMI in case they are not present in the HMI capabilities cache file (hmi_capabilities_cache.json) and
+--  the default capabilities
+
+-- Preconditions:
+-- 1  Value of HMICapabilitiesCacheFile parameter is defined (hmi_capabilities_cache.json) in smartDeviceLink.ini file
+-- 2. HMI capabilities cache file (hmi_capabilities_cache.json) doesn't exist on file system
+-- 3. One of HMI capabilities (VR/TTS/RC/UI etc) is missing in HMICapabilities file (stored in hmi_capabilities.json)
+-- 4. SDL and HMI are started
+-- 5. HMI does not provide one of HMI capabilities (VR/TTS/RC/UI etc)
+-- Sequence
+-- 1. Ignition OFF/ON cycle performed
+--  a. SDL sends appropriate HMI capabilities (VR/TTS/RC/UI/Buttons/VehicleInfo etc) request to HMI
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local common = require('test_scripts/Capabilities/PersistingHMICapabilities/common')
+
+--[[ Local Variables ]]
+local capabilitiesMap = {
+  UI = {
+    GetLanguage = { "language" },
+    GetSupportedLanguages = { "languages" },
+    GetCapabilities = { "displayCapabilities" }
+    },
+  VR = {
+    GetLanguage = { "language" },
+    GetSupportedLanguages = { "languages" },
+    GetCapabilities = { "vrCapabilities" }
+    },
+  TTS = {
+    GetLanguage = { "language" },
+    GetSupportedLanguages = { "languages" },
+    GetCapabilities = { "vrCapabilities" }
+    },
+  Buttons = {
+    GetCapabilities = { "capabilities" }
+  },
+  VehicleInfo = {
+    GetVehicleType = { "vehicleType" }
+  }
+}
+
+--[[ Local Functions ]]
+local function updateHMICaps(pMod, pRequest)
+  local hmiDefaultCap = common.getDefaultHMITable()
+  for key, _ in pairs (hmiDefaultCap) do
+    if key == pMod then
+      hmiDefaultCap[pMod][pRequest] = nil
+      if not pMod == "Buttons" then
+        hmiDefaultCap[pMod].IsReady.params.available = true
+      end
+    end
+  end
+  return hmiDefaultCap
+end
```

Comment:
@VjKlepikov What the loop for here? 
What purpose of removing of section `hmiDefaultCap[pMod][pRequest]`?

```suggestion
local function updateHMICaps(pMod, pRequest, pCap)
  local hmiCap = common.getDefaultHMITable()
  hmiDefaultCap[pMod][pRequest][pCap] = nil
  if not pMod == "Buttons" then
    hmiDefaultCap[pMod].IsReady.params.available = true
  end
  return hmiCap
end
```

Reply:
Done (see [15490e2](https://github.com/LuxoftSDL/sdl_atf_test_scripts/pull/29/commits/15490e2165ffad915c784dee5ffdc6665fc428b8) )